amr-wind/ocean_waves/relaxation_zones/waves2amr_ops.H

```diff
@@ -457,20 +439,61 @@ struct InitDataOp<W2AWaves>
         // Interpolate to MultiFabs (one level at a time)
         interp_to_mfab::interp_eta_to_levelset_multifab(
             wdata.n0, wdata.n1, wdata.dx0, wdata.dx1, wdata.zsl,
-            wdata.sp_eta_vec, levelset(level), problo, dx);
+            wdata.sp_eta_vec, ow_levelset(level), problo, dx);
         interp_to_mfab::interp_velocity_to_multifab(
             wdata.n0, wdata.n1, wdata.dx0, wdata.dx1, indvec, wdata.hvec,
-            wdata.sp_u_vec, wdata.sp_v_vec, wdata.sp_w_vec, velocity(level),
+            wdata.sp_u_vec, wdata.sp_v_vec, wdata.sp_w_vec, ow_velocity(level),
             problo, dx);
-        // Zero velocity in pure liquid cells
-        postprocess_velocity_mfab_liquid(velocity(level), levelset(level), dx);
-
-        // Copy to ow fields for future interpolation
-        auto& ow_levelset = sim.repo().get_field("ow_levelset");
-        auto& ow_velocity = sim.repo().get_field("ow_velocity");
-        amrex::MultiFab::Copy(ow_levelset(level), levelset(level), 0, 0, 1, 3);
-        amrex::MultiFab::Copy(
-            ow_velocity(level), velocity(level), 0, 0, AMREX_SPACEDIM, 3);
+        // Zero velocity in pure air cells
+        postprocess_velocity_mfab_liquid(
+            ow_velocity(level), ow_levelset(level), dx);
+
+        // Populate flow fields according to intended forcing and init setup
+        for (amrex::MFIter mfi(levelset(level)); mfi.isValid(); ++mfi) {
```

Comment:
Can we make this a fused MF now? That will make it work with openmp out of the box and better on GPU in general.

Reply:
yep, I should add that to this.